projboard/tests/conftest.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from projboard.models.article import Article, User, View_Article, Like, Subject
+
+
+TITLE = "Article Test"
+CONTENT = "I'm not gonna write all that!"
+
+
+@pytest.fixture
+def create_subjects():
+    sup = Subject(name='N')
+    sup1 = Subject(name='R')
+    sup.save()
+    sup1.save()
+    return [sup, sup1]
+
+
+@pytest.fixture
+def create_user():
+    user = User(name="my_username", nickname="my_nickname", email="my_email@gmail.com", password="my_password")
+    user1 = User(name="John", nickname="J99", email="John@gmail.com", password="John123")
+    user.save()
+    user1.save()
+    return [user, user1]
+
+
+@pytest.fixture
+def create_articles(create_user, create_subjects):
+    articles_list = []
+    for i in range(len(create_user)):
+        article = Article(user_id=create_user[i], title=TITLE, content=CONTENT, subject_id=create_subjects[i])
+        article.save()
+        articles_list.append(article)
+
+    return articles_list
+
+
+@pytest.fixture
+def create_like(create_user, create_articles):
+    like = Like(user_id=create_user[0], article_id=create_articles[0])
+    like.save()
+    return like
+
+
+@pytest.fixture
+def create_view(create_user, create_articles):
+    view_article = View_Article(user_id=create_user[0], article_id=create_articles[0])
+    view_article.save()
+    return view_article
+
+
+@pytest.fixture
+def generate_subject():
+    subject = Subject.create_and_get_subject(name="yael")
+    subject.save()
+    return subject
+
+
+@pytest.fixture
+def generate_subjects(generate_subject):
```

Comment:
@yaelnoyman1 I need your help here, I didn't really understood what's the difference between `generate_subject` and `generate_subjects`

Reply:
generate_subject is the creating one subject and generate_subjects is for creating two subjects for the function test_get_list_subjects_names

Reply:
We can keep the one with multi subjects and just edit the test function that use that with the one subject

Reply:
@Anas-mhameed I kept `subjects` function, so we need more test for `create_and_get_subject`, or we just change `create_and_get_subject` to `get_subject`, wdyt @yaelnoyman1 ?

Reply:
I assume that @yaelnoyman1 used the generate_subject in the test file of the subject model , so once you delete this fixture you should also change the functions that used this fixture .
This was my intent.

Reply:
@yaelnoyman1  Like to divide `create_and_get_subject` to `get_subject` and to `create_subject` (already existed), and then we just add to test_subject `test_get_subject` because create&test create subject are already existed

Reply:
We talked in Slack, And we decided to keep the function "creae_and_get_suject" as it is.
avi, You got your answer about "Why do we need both create_subjects and generate_subjects?"

Reply:
@avi-biton

Reply:
The `create_and_get_subject` was not the issue for me, I just wondered about 2 fixtures doing the same.  
I  see you narrowed it to one fixture called `subjects` so I am fine with it.